### ISSUE TITLE
Problem with image targets

### ISSUE THREAD

**Issue by mattagape** [sphinx-doc/sphinx#8819]
Given the following content near the start of an `.rst` file:

```
.. start-badges
 
 .. image:: https://github.com/UCL/TLOmodel/workflows/CI/badge.svg?branch=master
     :alt: GitHub Actions Build Status
     :target: https://github.com/UCL/TLOmodel/actions?query=workflow%3ACI
 
.. end-badges
```

On Github, the HTML generated for this link is:

```
<a href="https://github.com/UCL/TLOmodel/actions?query=workflow%3ACI">
 <img alt="GitHub Actions Build Status" src="https://github.com/UCL/TLOmodel/workflows/CI/badge.svg" style="max-width:100%;">
 </a>
```

But Sphinx generates this:

```
<a class="reference external image-reference" href="https://github.com/UCL/TLOmodel/actions?query=workflow%3ACI">
    <img alt="GitHub Actions Build Status" src="https://github.com/UCL/TLOmodel/workflows/CI/badge.svg" /></a>
```

i.e. we have an extra, unwanted `class="reference external image-reference"` attribute. What *should* be displayed is an image of the CI badge which, when clicked, goes to the current state of the CI build. But what happens on Sphinx is that the image is not displayed, and a question mark icon is instead. This can be clicked.

This is using Python 3.8.5 on a Mac. The command I use to generate the docs is `tox -e docs`

sphinx-build 2.4.4, sphinx-apidoc 2.4.4, tox version 3.21.3,

**Comment by tk0miya** [comment]
It seems the image on https://github.com/UCL/TLOmodel/workflows/CI/badge.svg?branch=master is not found now. I tried to reproduce this trouble using https://github.com/sphinx-doc/sphinx/workflows/CI/badge.svg?branch=master, and it works fine to me. Could you check the URL and try again please?

**Comment by mattagape** [comment]
Hi there.

I am sorry but the TLOmodel is a private repo, which is probably why it doesn't appear for you. I just checked and I can get to https://github.com/UCL/TLOmodel/workflows/CI/badge.svg?branch=master in my browser to display the badge OK. But the badge is still not displaying in my HTML generated by Sphinx. Perhaps a Github issue?

I tried using your link instead and it worked as expected.

Very strange!
